### PR TITLE
Implement triggers

### LINKED ISSUE #267: Triggers in foralls are not supported

### PR DESCRIPTION
**Pull request by vakaras**
Fixes #267

### CODE REVIEW

File: prusti-viper/src/encoder/spec_encoder.rs [hunk 1]
```diff
@@ -123,17 +124,14 @@ impl<'p, 'v: 'p, 'tcx: 'v> SpecEncoder<'p, 'v, 'tcx> {
         vir::LocalVar::new(var_name, vir::Type::Int)
     }
 
-    fn encode_trigger(&self, trigger: &typed::Trigger) -> vir::Trigger {
+    fn encode_trigger(&self, trigger: &typed::Trigger) -> Result<vir::Trigger, EncodingError> {
```
Comment (fpoli) [inline prusti-viper/src/encoder/spec_encoder.rs]: We have `EncodingResult<vir::Trigger>`, if you wish.

**Comment by Aurel300** [comment]
Slightly unrelated to the PR, but I see graphviz related changes – would it be useful to have a section on using the visualisations in the dev guide?

**Comment by vakaras** [comment]
> Slightly unrelated to the PR, but I see graphviz related changes – would it be useful to have a section on using the visualisations in the dev guide?

Yes, I think it would be very useful to have instructions on how to get visualizations and how to use them for debugging.

**Review by vakaras — COMMENTED** [review]
.

**Pull request merged**